packages/labs/src/components/popover/popover2.tsx

```diff
@@ -0,0 +1,527 @@
+                >
+                    {this.renderPopper(children.content)}
+                </Overlay>
+            </Manager>
+        );
+    }
+
+    public componentDidMount() {
+        this.updateDarkParent();
+    }
+
+    public componentWillReceiveProps(nextProps: IPopover2Props) {
+        super.componentWillReceiveProps(nextProps);
+
+        if (nextProps.isOpen == null && nextProps.isDisabled && !this.props.isDisabled) {
+            // ok to use setOpenState here because isDisabled and isOpen are mutex.
+            this.setOpenState(false);
+        } else if (nextProps.isOpen !== this.props.isOpen) {
+            // propagate isOpen prop directly to state, circumventing onInteraction callback
+            // (which would be invoked if this went through setOpenState)
+            this.setState({ isOpen: nextProps.isOpen});
+        }
+    }
+
+    public componentWillUpdate(_: IPopover2Props, nextState: IPopover2State) {
+        if (!this.state.isOpen && nextState.isOpen) {
+            this.isContentMounting = true;
+            Utils.safeInvoke(this.props.popoverWillOpen);
+        } else if (this.state.isOpen && !nextState.isOpen) {
+            Utils.safeInvoke(this.props.popoverWillClose);
+        }
+    }
+
+    public componentDidUpdate() {
+        this.updateDarkParent();
+    }
+
+    public componentWillUnmount() {
+        super.componentWillUnmount();
+    }
+
+    private updateDarkParent() {
+        if (!this.props.inline) {
+            const hasDarkParent = this.targetElement.closest(`.${Classes.DARK}`) != null;
+            this.setState({ hasDarkParent });
+        }
+    }
+
+    private renderPopper(content: JSX.Element) {
+        const { inline, interactionKind, modifiers } = this.props;
+        const popoverHandlers: React.HTMLAttributes<HTMLDivElement> = {
+            // always check popover clicks for dismiss class
+            onClick: this.handlePopoverClick,
+        };
+        if ((interactionKind === PopoverInteractionKind.HOVER)
+            || (inline && interactionKind === PopoverInteractionKind.HOVER_TARGET_ONLY)) {
+            popoverHandlers.onMouseEnter = this.handleMouseEnter;
+            popoverHandlers.onMouseLeave = this.handleMouseLeave;
+        }
+
+        const popoverClasses = classNames(Classes.POPOVER, {
+            [Classes.DARK]: this.props.inheritDarkTheme && this.state.hasDarkParent,
+            [Classes.MINIMAL]: this.props.minimal,
+        }, this.props.popoverClassName);
+
+        const isArrowEnabled = !this.props.minimal
+            // omitting `arrow` from `modifiers` uses Popper default, which does show an arrow.
+            && (modifiers.arrow == null || modifiers.arrow.enabled);
+        const allModifiers: PopperModifiers = {
+            ...modifiers,
+            arrowOffset: {
+                enabled: isArrowEnabled,
+                fn: arrowOffsetModifier,
+                order: 510, // arrow is 500
```

Comment:
i mean yeah, straight from the popper docs: https://popper.js.org/popper-documentation.html#modifiers..arrow.order